js/queue/queue-data-structure.js

```diff
@@ -9,18 +9,35 @@ class QueueDataStructure {
   }
 
   canEnqueue() {
-    // ... your code goes here
+    if (this.queueControl.length < this.MAX_SIZE) {
+      return true
+    } else {
+      return false
+    }
   }
 
   isEmpty() {
-    // ... your code goes here
+    if (this.queueControl.length <= 0) {
+      return true
+    } else {
+      return false
+    }
   }
 
   enqueue(item) {
-    // ... your code goes here
+    if (this.canEnqueue()) {
+      this.queueControl.unshift(item)
+      return this.queueControl
+    } else {
+      return "Queue Overflow"
+    }
   }
 
   dequeue() {
-    // ... your code goes here
+    if (this.isEmpty()) {
+      return "Queue Underflow"
+    } else {
+      return this.queueControl.shift()
```

Comment:
```suggestion
      return this.queueControl.pop()
```
Recuerda que el método pop() elimina el último elemento de un array y lo devuelve.